pkg/util/dynamichelper/dynamichelper_test.go

```diff
@@ -645,3 +653,89 @@ func TestMakeURLSegments(t *testing.T) {
 		})
 	}
 }
+
+func TestEnsure(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
+	manager := mock_dynamichelper.NewMockGVRInterface(controller)
+
+	manager.EXPECT().Resolve(
+		gomock.Eq("Event"),
+		gomock.Any(),
+	).DoAndReturn(func(groupKind, optionalVersion string) (*schema.GroupVersionResource, error) {
+		return &schema.GroupVersionResource{
+			Group:    "",
+			Version:  "v1",
+			Resource: "event",
+		}, nil
+	})
+
+	h := http.Header{}
+	h.Add("Content-Type", "application/yaml")
+	var d kruntime.Object = &corev1.Event{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Event",
+			APIVersion: "v1",
+		},
+	}
+	// Prepare the encoder
+	serializer := kjson.NewSerializerWithOptions(
+		kjson.DefaultMetaFactory, scheme.Scheme, scheme.Scheme,
+		kjson.SerializerOptions{Yaml: true},
+	)
+	yaml := scheme.Codecs.CodecForVersions(serializer, nil, schema.GroupVersions(scheme.Scheme.PrioritizedVersionsAllGroups()), nil)
+
+	buf := new(bytes.Buffer)
+	err := yaml.Encode(d, buf)
+	if err != nil {
+		t.Errorf("Failed to encode document: %s", err.Error())
+	}
+
+	frc := &fake.RESTClient{
+		GroupVersion:         schema.GroupVersion{Group: "", Version: "v1"},
+		NegotiatedSerializer: scheme.Codecs.WithoutConversion(),
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+			switch req.Method {
+			case "GET":
+
+				switch req.URL.Path {
+				case "/api/v1/event":
+					return &http.Response{
+						StatusCode: http.StatusOK,
+						Header:     h,
+						Body:       io.NopCloser(buf),
+					}, nil
+				}
```

Comment:
why nested switch statement instead of a simpler 
```go
if req.Method == http.MethodGET && req.URL.Path == "/api/v1/event" {
    return ... 
}
``` 
?
btw please prefer the constant from the http file instead of write "GET" yourself